evmbin/src/main.rs

```diff
@@ -74,24 +82,40 @@ pub fn run_vm(params: ActionParams) -> ExecutionResults {
 	let mut ext = ext::FakeExt::default();
 
 	let start = Instant::now();
-	let gas_left = vm.exec(params, &mut ext).finalize(ext).expect("OK");
+	let gas_left = vm.exec(params, &mut ext).finalize(ext);
 	let duration = start.elapsed();
 
-	ExecutionResults {
-		gas_used: initial_gas - gas_left,
-		output: Vec::new(),
-		time: duration,
+	match gas_left {
+		Ok(gas_left) => ExecutionResults::Ok {
+			gas_used: initial_gas - gas_left,
+			output: Vec::new(),
+			time: duration,
+		},
+		Err(e) => ExecutionResults::Error {
+			error: e,
+			time: duration,
+		},
 	}
 }
 
 /// VM execution results
-pub struct ExecutionResults {
-	/// Used gas
-	pub gas_used: U256,
-	/// Output as bytes
-	pub output: Vec<u8>,
-	/// Time Taken
-	pub time: Duration,
+pub enum ExecutionResults {
+	/// Execution finished correctly
+	Ok {
+		/// Used gas
+		gas_used: U256,
+		/// Output as bytes
+		output: Vec<u8>,
+		/// Time Taken
+		time: Duration,
+	},
+	/// Error
+	Error {
```

Comment:
Should be `Err` for parallel to `Result`.
And the type should be called `ExecutionResult`. Then it could just be `type ExecutionResult = Result<Success, Failure>` with type definitions for the enum struct variants.